mini/fibonacci_test.go

```diff
@@ -0,0 +1,122 @@
+package mini
+
+import (
+	"testing"
+
+	"github.com/awalterschulze/gominikanren/micro"
+	"github.com/awalterschulze/gominikanren/sexpr/ast"
+)
+
+// peano numbers and extralogical convenience functions
+var zero = ast.NewInt(0)
+var one = makenat(1)
+
+/*
+   (define (succ x y)
+       (== y `(,x)))
+*/
+func succ(prev, next *ast.SExpr) micro.Goal {
```

Comment:
These nat stuff is nice enough to put into a `nat` folder.
Not in this pull request but in a following one, what do you think?

Reply:
If we put everything in its separate folder sure; if we put all the examples in one (which was what I had in mind before) then it doesnt really make sense to single these out.

Reply:
I just think `nat` would be useful to a lot of use cases.